app/ui/src/main/java/com/fsck/k9/activity/setup/AccountSetupBasics.java

```diff
@@ -423,15 +445,44 @@ private void onManualSetup() {
         finish();
     }
 
+    private void onAccountManagerSetup(String email) {
+        if (mAccount == null) {
+            mAccount = Preferences.getPreferences(this).newAccount();
+            mAccount.setChipColor(AccountCreator.pickColor(this));
+        }
+        mAccount.setName(getOwnerName());
+        mAccount.setEmail(email);
+
+        ServerSettings storeServer = new ServerSettings(Protocols.IMAP, "imap.gmail.com", -1,
```

Comment:
These server constants work well even for Google for Domains but maybe they should be moved somewhere else (to the OAuth provider?)

Reply:
Part of the complexity was working out how Google specific to make it - quite a few of the providers use XOAUTH2 now. Ideally there'd be no reference to Google anywhere except in a configuration system.

Reply:
Good idea, I've seen a ticket about Yahoo (#1698), do you know if it's working and how to test it? (If we need to register the app etc...)

Reply:
We do, so it doesn't and it's basically impossible to get an API key - I tried tweeting a bigwig and Yahoo but it didn't go anywhere. 

I think getting it working with Microsoft might be doable.

However, the number of requests for those is much smaller. I'd try to ship Google and get feedback - it'll solve 90% of people's demands - you can always refactor later.